### PR TITLE
gh-122332: Fix missing `NULL` check in `asyncio.Task.get_coro`

### PR DESCRIPTION
**Pull request by ZeroIntensity**
* Issue: gh-122332

### CODE REVIEW

File: Modules/_asynciomodule.c [hunk 1]
```diff
@@ -2500,7 +2500,12 @@ static PyObject *
 _asyncio_Task_get_coro_impl(TaskObj *self)
 /*[clinic end generated code: output=bcac27c8cc6c8073 input=d2e8606c42a7b403]*/
 {
-    return Py_NewRef(self->task_coro);
+    if (self->task_coro)
+    {
```
Comment (JelleZijlstra) [inline Modules/_asynciomodule.c]:
```suggestion
    if (self->task_coro) {
```

**Review by JelleZijlstra — COMMENTED** [review]
Could you add a unit test for this? Something that would currently segfault.

**Comment by ZeroIntensity** [comment]
> Could you add a unit test for this? Something that would currently segfault.

Added it, it's basically a translation of my reproducer in #122332.

**Review by Eclips4 — APPROVED** [review]
Thanks!

**Review by kumaraditya303 — APPROVED** [review]

**Pull request merged**